framework/include/utils/InputParameters.h

```diff
@@ -350,14 +422,36 @@ class InputParameters : public libMesh::Parameters
   bool isCommandLineParameter(const std::string & name) const;
 
   /**
-   * @return The command line syntax for the parameter \p name
+   * @return Queries for the command line metadata for the parameter \p name
+   *
+   * Will return an empty optional if the parameter is not a command line param.
    */
-  const std::vector<std::string> & getCommandLineSyntax(const std::string & name) const;
+  std::optional<InputParameters::CommandLineMetadata>
+  queryCommandLineMetadata(const std::string & name) const;
 
   /**
-   * @return The command line argument type for the parameter \p name
+   * @return The command line metadata for the parameter \p name.
+   */
+  const InputParameters::CommandLineMetadata &
+  getCommandLineMetadata(const std::string & name) const;
+
+  /**
+   * Class that is used as a parameter to commandLineParamSet() that allows only
+   * the CommandLine to set that a parmeter is set by the command line
+   */
+  class CommandLineParamSetKey
+  {
+    friend class CommandLine;
+    FRIEND_TEST(InputParametersTest, commandLineParamSetNotCLParam);
+    CommandLineParamSetKey() {}
+    CommandLineParamSetKey(const CommandLineParamSetKey &) {}
+  };
```

Comment:
Uhm, we have `Passkey.h` with a template for pass keys...

Reply:
Doesn't work for multiple friends, nor gtest friends